### PR TITLE
feat: multipart support

### PR DESCRIPTION
**Pull request by charlieforward9**
Tried to acknowledge the contributions of the previous contributors to for getting multipart geometries working. Closes https://github.com/robertleeplummerjr/Leaflet.glify/issues/42 and https://github.com/robertleeplummerjr/Leaflet.glify/pull/66 and https://github.com/robertleeplummerjr/Leaflet.glify/pull/158

Co-authored-by: Kanahiro Iguchi <mediterranean1769@gmail.com> @Kanahiro 
Co-authored-by: Liam O'flynn <lpoflynn@protonmail.ch> @quorth0n

Includes Prettier formatting

### CODE REVIEW

**Comment by charlieforward9** [comment]
@robertleeplummerjr I tested this using the Antarctica.geojson file and it seems to render "good enough" for now. A strange overlap issue could be addressed down the road, when someones use case requires it. Can be merged. I will be depending on this PR branch till then.

Screenshot 2024-11-25 at 3 40 19 PM

Also willing to take responsibility for managing this repo if you find yourself too busy. Add me as a contributor if you'd like a teammate in this. (Still highly undereducated in the GL space, but managing repos/packages is a bit more straightforward)

**Comment by charlieforward9** [comment]
I am having a problem with the click and popup handlers. It doesnt seem related to this because the example works fine, but curious to explore deeper into this to resolve.

File: README.md
Comment (robert-lore) [inline README.md:1]: Why all the changes to the readme?

Comment (robertleeplummerjr) [reply README.md:1]: Sorry I was logged in with my work account.

Comment (charlieforward9) [reply README.md:1]: I had prettier formatting enabled, which made all of these changes automatically. I removed it for this file for the time being, and ammended the commit. @robertleeplummerjr

**Comment by charlieforward9** [comment]
@robertleeplummerjr this should be ready to merge and all related issues/pr's can be closed.

Just a reminder there are some stragler issues to close from my previous PR, see [this comment](https://github.com/robertleeplummerjr/Leaflet.glify/pull/156#issuecomment-2177205250)

File: README.md
Comment (robertleeplummerjr) [reply README.md:1]: No need to turn them off.  I was just curious as to why they were done.

Comment (charlieforward9) [reply README.md:1]: I turned it back on for my workspace, so the next time one of these contributions comes around I'll leave the changes in place.

**Review by robert-lore — APPROVED** [review]

**Comment by charlieforward9** [comment]
@robert-lore @robertleeplummerjr  thank you for the merge. Do you plan to make a release and close the resolved issues? I can help you manage this repo if you find yourself without time.

**Comment by robert-lore** [comment]
I'm installing it on my mac and was having trouble.

**Comment by charlieforward9** [comment]
@robert-lore  Were they from my contributions or your machine?

**Comment by arvii** [comment]
not yet available in https://yarnpkg.com/package?q=leaflet.glify&name=leaflet.glify  ? Thank you so much

**Comment by arvii** [comment]
@charlieforward9  does your fix support combination of Polygon and MultiPolygon on a FeatureCollection? 🙇 
Screenshot 2024-12-09 at 9 35 00 PM

**Comment by robert-lore** [comment]
Just removed canvas, which was causing my install issue, and published.

**Comment by charlieforward9** [comment]
> @charlieforward9 does your fix support combination of Polygon and MultiPolygon on a FeatureCollection? 🙇 

Worth taking a look and letting me know. I just took the contributions of @quorth0n and @Kanahiro and applied them to the latest branch, so please let me know if you have any issues.

**Comment by trafficonese** [comment]
> @charlieforward9 does your fix support combination of Polygon and MultiPolygon on a FeatureCollection? 🙇

@arvii , yes it works also for a combination of polygons and multipolygons.
image

**Comment by arvii** [comment]
> > @charlieforward9 does your fix support combination of Polygon and MultiPolygon on a FeatureCollection? 🙇
> 
> @arvii , yes it works also for a combination of polygons and multipolygons. image

Hi @trafficonese Thank you so much for checking also. Could you please send me the sample data you tested? I want to verify if my format is correct. I tested it with a combination of Polygon and MultiPolygon in a GeoJSON FeatureCollection, but only the Polygon rendered. When I tested a GeoJSON Feature with a single MultiPolygon, it rendered correctly. Thank you so much! 🙇

**Comment by trafficonese** [comment]
@arvii It's a very simple Featurecollection:

```
const data: FeatureCollection<Geometry> = {
  type: "FeatureCollection",
  features: [
    {
      type: "Feature",
      properties: { name: "Polygon 1" },
      geometry: {
        type: "Polygon",
        coordinates: [
          [
            [102.0, 0.0],
            [103.0, 0.0],
            [103.0, 1.0],
            [102.0, 1.0],
            [102.0, 0.0],
          ],
        ],
      },
    },
    {
      type: "Feature",
      properties: { name: "MultiPolygon 1" },
      geometry: {
        type: "MultiPolygon",
        coordinates: [
          [
            [
              [100.0, 0.0],
              [101.0, 0.0],
              [101.0, 1.0],
              [100.0, 1.0],
              [100.0, 0.0],
            ],
          ],
          [
            [
              [100.2, 0.2],
              [100.8, 0.2],
              [100.8, 0.8],
              [100.2, 0.8],
              [100.2, 0.2],
            ],
          ],
        ],
      },
    },
    {
      type: "Feature",
      properties: { name: "Polygon 2" },
      geometry: {
        type: "Polygon",
        coordinates: [
          [
            [104.0, 0.0],
            [105.0, 0.0],
            [105.0, 1.0],
            [104.0, 1.0],
            [104.0, 0.0],
          ],
        ],
      },
    },
    {
      type: "Feature",
      properties: { name: "MultiPolygon 2" },
      geometry: {
        type: "MultiPolygon",
        coordinates: [
          [
            [
              [102.0, 2.0],
              [103.0, 2.0],
              [103.0, 3.0],
              [102.0, 3.0],
              [102.0, 2.0],
            ],
          ],
          [
            [
              [102.2, 2.2],
              [102.8, 2.2],
              [102.8, 2.8],
              [102.2, 2.8],
              [102.2, 2.2],
            ],
          ],
        ],
      },
    },
  ],
};
```

**Pull request merged**